### PR TITLE
Bundle changes and restructure

### PR DESCRIPTION
**Pull request by wincent**
This is a combination of the commits in existing PRs:

- https://github.com/wolffiex/shellbot/pull/7
- https://github.com/wolffiex/shellbot/pull/8
- https://github.com/wolffiex/shellbot/pull/9
- https://github.com/wolffiex/shellbot/pull/10 ← see this one for screenshots
- https://github.com/wolffiex/shellbot/pull/11

Plus a few extra commits:

- d98d049 chore: try GPT-4o
- d95b524 refactor: move Neovim files into `lua/` subdirectory
- 0682fbd feat: add syntax file
- 3201989 feat: add `:ChatGPT` command (later renamed to `:Shellbot`)
- a089bc6 docs: add instructions for installation
- 0809fdb feat: add an ftplugin file
- a24dafd refactor: move settings from module into ftplugin
- 321ea3e refactor: replace deprecated `nvim_buf_set_option()` call
- c934b21 refactor: move mappings into ftplugin
- 7a92356 feat: add ability to set model at runtime via env var
- f848d28 fix: suppress system prompt for "o1-preview" and "o1-mini" models
- f63da96 refactor!: rename `:ChatGPT` to `:Shellbot`
- c9f2d71 chore: default Anthropic to "claude-3-5-sonnet-20241022"
- eb6d184 docs: document new environment variables
- 78a6114 chore: bump default Anthropic model to claude-3-7-sonnet-20250219
- 4f06149 feat: teach `chatbot()` to take an `env` table
- 84b1c35 chore: bump OpenAI model to GPT-4.1
- f8470e5 chore: bump models to latest

Basically, moved a bunch a stuff of around to make the repo follow the standard directory layout for Vim plugins, which means that we can ship a syntax file, set up the ~`:ChatGPT`~ `:Shellbot` command for users automatically etc, as discussed in #10. I normally wouldn't bundle all this stuff into a single PR, but GitHub doesn't support stacked PRs, and the file moves will otherwise be in conflict with the previous PRs. I'm going to close those ones.

### CODE REVIEW

**Comment by wincent** [comment]
Superseded by:

- https://github.com/wolffiex/shellbot/pull/12

**Comment by wincent** [comment]
Whoops. Accidentally closed this right after opening it. Reopening.
